.github/workflows/wait_for_required_workflows.yml

```
@@ -0,0 +1,73 @@
+name: Wait for all required workflows to pass
+
+on: [pull_request, push]
```

```suggestion
on:
  pull_request:
    branches:
      - main
```

Do we need to run this on every push to the repository? Maybe only do it on pushes to `main`, but also since we require branches to be up to date, we can only check PRs I think